gulp/tasks/copy.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+var gulp = require( 'gulp' );
+var $ = require( 'gulp-load-plugins' )();
+var config = require( '../config' ).copy;
+var handleErrors = require( '../utils/handleErrors' );
+var browserSync = require( 'browser-sync' );
+
+gulp.task( 'copy:files', function() {
+  return gulp.src( config.files.src )
+    .pipe( $.changed( config.files.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( gulp.dest( config.files.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:legacy', function() {
+  return gulp.src( config.legacy.src )
+    .pipe( $.changed( config.legacy.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( gulp.dest( config.legacy.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:icons', function() {
+  return gulp.src( config.icons.src )
+    .pipe( $.changed( config.icons.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( gulp.dest( config.icons.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:pdfcss', function() {
+  return gulp.src( config.pdfcss.src )
+    .pipe( $.changed( config.pdfcss.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( gulp.dest( config.pdfcss.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:pdffonts', function() {
+  return gulp.src( config.pdffonts.src )
+    .pipe( $.changed( config.pdffonts.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( gulp.dest( config.pdffonts.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:vendorcss', function() {
+  return gulp.src( config.vendorcss.src )
+    .pipe( $.changed( config.vendorcss.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( $.replace(
+      /url\(".\/ajax-loader.gif"\)/ig,
+      'url("/static/img/ajax-loader.gif")'
+    ) )
+    .pipe( gulp.dest( config.vendorcss.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:vendorimg', function() {
+  return gulp.src( config.vendorimg.src )
+    .pipe( $.changed( config.vendorimg.dest ) )
+    .on( 'error', handleErrors )
+    .pipe( gulp.dest( config.vendorimg.dest ) )
+    .pipe( browserSync.reload( {
+      stream: true
+    } ) );
+} );
+
+gulp.task( 'copy:vendorjs', function() {
```

Comment:
@jimmynotjim  I was thinking ( I think your approach is better for the reasons you mentioned ) :

```
function copyFiles ( src, destination ) {

  return gulp.src( src )
    .pipe( changed( destination ) )
    .on( 'error', handleErrors )
    .pipe( gulp.dest( destination ) )
    .pipe( browserSync.reload( {
      stream: true
    } ) );

}

gulp.task( 'copy:files', copyFiles.bind( undefined, config.files.src, config.files.dest ) );

gulp.task( 'copy:legacy', copyFiles.bind( undefined, config.legacy.src, config.legacy.dest ) );

```